addons/cookoff/functions/fnc_cookOff.sqf

```diff
@@ -113,11 +113,20 @@ if (local _vehicle) then {
 
         [_vehicle, _fnc_FlameEffect, 12] call _fnc_FlameEffect; // recursive function
 
+        private _randomPosition = _vehicle getPos [100, random 360];
+
         {
-            if (local _x && {!(_x call EFUNC(common,isPlayer))}) then {
-                _x action ["Eject", _vehicle];
+            _x leaveVehicle _vehicle;
+            
+            if (leader _x == _x) then {
+                private _wp = group _x addWaypoint [_randomPosition, 0, 1];
```

Comment:
Needs optional parameter:
- index (optional): Number - can be used to insert a waypoint in between existing waypoints

https://community.bistudio.com/wiki/addWaypoint

otherwise the waypoint will be added as the last waypoint, not as the current one!
I think it should be always 0 or `currentWaypoint _group` or `currentWaypoint _group -1` or something similar